components/script/dom/range.rs

```diff
@@ -442,10 +450,19 @@ impl RangeMethods for Range {
         }
         // Step 3.
         let (this_point, other_point) = match how {
-            RangeConstants::START_TO_START => (&self.start, &other.start),
-            RangeConstants::START_TO_END => (&self.end, &other.start),
-            RangeConstants::END_TO_END => (&self.end, &other.end),
-            RangeConstants::END_TO_START => (&self.start, &other.end),
+            RangeConstants::START_TO_START => (
+                self.abstract_range().start(),
+                other.abstract_range().start(),
```

Comment:
Could you add a `start()` and `end()` helper to avoid having to use so much repetition?